runners/google-cloud-dataflow-java/src/main/java/org/apache/beam/runners/dataflow/util/PackageUtil.java

```diff
@@ -456,7 +477,36 @@ public static PackageAttributes forFileToStage(File source, String stagingPath)
       target.setName(uniqueName);
       target.setLocation(resourcePath);
 
-      return new AutoValue_PackageUtil_PackageAttributes(source, target, size, hash);
+      return new AutoValue_PackageUtil_PackageAttributes(source, null, target, size, hash);
+    }
+
+    public static PackageAttributes forBytesToStage(byte[] bytes, String targetName, String stagingPath)
+        {
+      long size;
+      String hash;
+      Hasher hasher = Hashing.md5().newHasher();
```

Comment:
hasher.putBytes(bytes)? instead of wrapping in a funnel/outputstream/...

Reply:
Done. (was copy pasta from file processing, to make it easy to see that it is identical)